### PR TITLE
Fix merge conflicts and a few minor issues in optionals.py

### PR DESCRIPTION
**Pull request by eliarbel**
### Summary
I'm not sure how common of a workflow it is to open a PR against a Mergify backport branch, but still: 
proposing conflict resolution and a few other minor fixes (e.g. broken Aer link and typos) against #13101. Possibly also a follow-up for #13098.

### CODE REVIEW

File: qiskit/utils/optionals.py [hunk 1]
```diff
@@ -27,32 +27,18 @@
...
+    `Qiskit Aer <https://qiskit.github.io/qiskit-aer/>`__ provides high-performance simulators for
     the quantum circuits constructed within Qiskit.
 
 .. py:data:: HAS_IBMQ
 
-    The :mod:`Qiskit IBMQ Provider <qiskit.providers.ibmq>` is used for accessing IBM Quantum
-    hardware in the IBM cloud.
->>>>>>> 254ba83dc (Fix documentation for utils.optionals data values (#13098))
+    The Qiskit IBMQ Provider was historically used for accessing IBM Quantum hardware in the IBM
+    cloud, but is now deprecated.
```
Comment (ElePT) [inline qiskit/utils/optionals.py]:
```suggestion
    cloud, but is now archived.
```

File: qiskit/utils/optionals.py [hunk 2]
```diff
@@ -27,32 +27,18 @@
...
 
 .. py:data:: HAS_IGNIS
 
-    :mod:`Qiskit Ignis <qiskit.ignis>` provides tools for quantum hardware verification, noise
-    characterization, and error correction.
+    Qiskit Ignis was historically used to provide tools for quantum hardware verification,
+    noise characterization, and error correction, but is now deprecated.
```
Comment (ElePT) [inline qiskit/utils/optionals.py]:
```suggestion
    noise characterization, and error correction, but is now archived.
```

**Review by ElePT — COMMENTED** [review]
I tend to push directly to the mergify branch, but I think a PR works too. The only downside to opening a PR is that with it, the backport requires merging 2 PRs instead of one, which consumes more CI resources. It might make more sense in cases like these where a 3rd person is fixing the conflicts and wants approval from the original author.

**Review by Eric-Arellano — APPROVED** [review]
Thanks for fixing!

**Pull request merged**